### ISSUE TITLE
Check for dead URLs in documentation files

### ISSUE THREAD

**Issue by per1234** [per1234/inolibbuglist#5]
arduino-ci-script already checks the library.properties `url` field for dead URLs.

Likely there are some good existing solutions that would make this fairly easy to do. I want to add this sort of check to my CI tests anyway so the effort will have multiple payoffs.

**Comment by TomasRoj** [comment]
I had a look and I found this python lib: https://scrapy.org/. Some guys on forum always said that its good for checking dead links. We can make some algorythm or something like that

**Comment by per1234** [comment]
My instinct is that the best solution would act on the local files after the repository was downloaded. I think this would be significantly faster for larger repositories than scanning the online repository. I get the feeling that would not be possible to do with scrapy.

**Comment by ArminJo** [comment]
Hi,
the playground.arduino.cc URL's appear to be case sensitive now which leads to hundred of "broken" links in the libraries.properties files.
E.g. https://playground.arduino.cc/code/time/  is invalid and links to the start page (no 404!) the new one must be https://playground.arduino.cc/Code/Time/ (captial C and capital T).
Can you use your influence and convince the responsible persons to switch back to case insensitve URLs, since most of the libraries have the same issue.
This would increase the quality of current library documentation a lot.
Thank you for your effort.
Armin

**Comment by per1234** [comment]
>the playground.arduino.cc URL's appear to be case sensitive

Interesting. I wasn't aware of that. The pages with incorrect case redirect to the Playground's home page so my check of the `url` field in library.properties didn't expose this issue. I don't think there is any way to differentiate a "good" redirect from a "bad" redirect. I could make the check not follow redirects, but that would result in a lot of false positives on the bad URL check. I'll have to give that some consideration. For this reason, it would be preferable if the Playground just gave a 404 error for the URLs with incorrect case.

>which leads to hundred of "broken" links in the libraries.properties files.

Where does the "hundred" figure come from? It would be useful to know the extent of the impact of this change, but I don't know how I would determine this.

>Can you use your influence and convince the responsible persons to switch back to case insensitve URLs

The truth is that I don't have any more influence over this situation than you do. In fact you might have more than me, since Arduino very much values the input from users. Although I very much still consider myself a user, I'm also now a recently added member of the Arduino team, which perhaps gives my voice a bit less weight in this sort of decision.

**Comment by ArminJo** [comment]
> which leads to hundred of "broken" links in the libraries.properties files.

OK, ok there are only 14 entries in the json:
```
      "website": "http://playground.arduino.cc/code/time",
      "website": "http://playground.arduino.cc/code/time",
      "website": "http://playground.arduino.cc/code/time",
      "website": "http://playground.arduino.cc/code/time",
      "website": "http://playground.arduino.cc/code/time",
      "website": "http://playground.arduino.cc/code/SmoothADC",
      "website": "http://playground.arduino.cc/code/SmoothADC",
      "website": "http://playground.arduino.cc/code/SmoothADC",
      "website": "http://playground.arduino.cc/code/SmoothADC",
      "website": "http://playground.arduino.cc/code/CI2C",
      "website": "http://playground.arduino.cc/code/CaptureTimer",
      "website": "http://playground.arduino.cc/code/CaptureTimer",
      "website": "http://playground.arduino.cc/code/CaptureTimer",
      "website": "http://playground.arduino.cc/code/SeqTimer",
```
And all but the 5 "time" ones of Paul are dead, even if you use capital letters.

**Comment by per1234** [comment]
>OK, ok there are only 14 entries in the json:

Good idea searching the library_index.json file! I hadn't thought of that.

>`"website": "http://playground.arduino.cc/code/time",`

I submitted a PR to fix this issue: https://github.com/PaulStoffregen/Time/pull/128

>`"website": "http://playground.arduino.cc/code/SmoothADC",`

This has already been fixed: https://github.com/SMFSW/SmoothADC/commit/4874d42a55877731911addeaca667a0afc946818

>`"website": "http://playground.arduino.cc/code/CI2C",`

This has already been fixed: https://github.com/SMFSW/cI2C/commit/52b3885bfc44de3b7ba99fb103e18e7601f6b807

>`"website": "http://playground.arduino.cc/code/CaptureTimer",`

This has already been fixed: https://github.com/SMFSW/CaptureTimer/commit/5262b47ddfc6391133501b6321b820d0ee9a1873

>`"website": "http://playground.arduino.cc/code/SeqTimer",`

This has already been fixed: https://github.com/SMFSW/SeqTimer/commit/a5466fb34255a50c3b6b88eb81181097b026b951

Note that the fixes only will apply to releases made after the time of the fix. The existing releases will continue to have the bad URL. However, people are very unlikely to use old library releases so the problem is not very serious.
